### PR TITLE
Add workspace module for azure

### PR DESCRIPTION
**Pull request by yuwzho**
##### SUMMARY

Azure workspace module which is used for container  monitoring like AKS

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by yuwzho** [comment]
@mattclay The Ansible CI  is not register to `OperationInsight` namespaces. Could you please help me to do the registeration? ref: [Resolve errors for resource provider registration](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-register-provider-errors)

**Comment by mattclay** [comment]
@yuwzho `Microsoft.OperationalInsights` has been registered and the tests restarted.

File: lib/ansible/modules/cloud/azure/azure_rm_workspace.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_workspace
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_workspace.py]: should probably be `azure_rm_loganalyticsworkspace` or something; "workspace" seems awfully generic

File: lib/ansible/modules/cloud/azure/azure_rm_workspace_facts.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_workspace_facts
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_workspace_facts.py]: ... and match whatever the module above ends up being called

File: lib/ansible/modules/cloud/azure/azure_rm_workspace.py [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+    - azure_tags
+
+author:
+    - "Yuwei Zhou (@yuwzho)"
+'''
+
+EXAMPLES = '''
+'''
+
+RETURN = '''
```
Comment (nitzmahone) [inline lib/ansible/modules/cloud/azure/azure_rm_workspace.py]: documented results keys don't appear to match actual results returned

**Review by nitzmahone — CHANGES_REQUESTED** [review]
just name bikeshedding and results docs, otherwise looks OK to me

**Comment by yuwzho** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@yungezz @zikalino review_need

File: lib/ansible/module_utils/azure_rm_common.py [hunk 4]
```diff
@@ -963,6 +966,19 @@ def monitor_client(self):
                                                             base_url=self._cloud_environment.endpoints.resource_manager)
         return self._monitor_client
 
+    @property
+    def log_analytics_client(self):
+        self.log('Getting log analytics client')
+        if not self._log_analytics_client:
+            self._log_analytics_client = self.get_mgmt_svc_client(LogAnalyticsManagementClient,
```
Comment (yungezz) [inline lib/ansible/module_utils/azure_rm_common.py:973]: no api version?

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            - The SKU of the workspace
+        choices:
+            - free
+            - standard
+            - premium
+            - unlimited
+            - per_node
+            - per_gb2018
+            - standalone
+        default: per_gb2018
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py:53]: why default?

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+    retention_in_days:
+        description:
+            - The workspace data retention in days.
+            - -1 means Unlimited retention for the C(unlimited) C(sku).
+            - 730 days is the maximum allowed for all other C(sku)s.
+    intelligence_packs:
+        description:
+            - Manage intelligence packs possible for this workspace.
+            - "Enable one pack by setting it to C(true). E.g. {'Backup': true}."
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py:62]: didn't get it, what's backup mean? are keys of the dict random?  or choices?

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+        Backup: true
+'''
+
+RETURN = '''
+id:
+    description: Workspace resource path.
+    type: str
+    returned: success
+    example: "/subscriptions/XXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXX/resourceGroups/foo/providers/Microsoft.OperationalInsights/workspaces/bar"
+location:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py:89]: return those info in main module?

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace_facts.py [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_loganalyticsworkspace_facts
+version_added: "2.8"
+short_description: Manage Azure Log Analytics workspaces.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace_facts.py]: Get facts of xxx

File: lib/ansible/module_utils/azure_rm_common.py (hunk 4 above)
Comment (yuwzho) [reply lib/ansible/module_utils/azure_rm_common.py:973]: no

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py (hunk 5 above)
Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py:53]: According to Azure CLI and docs, they use the per_gb2018 as default SKU, which is used only when creation.

File: lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py (hunk 6 above)
Comment (yuwzho) [reply lib/ansible/modules/cloud/azure/azure_rm_loganalyticsworkspace.py:62]: I didn't find an official docs here. My understanding this is a list contains many configurations of the workspace. `backup` is one of the configurations. All the list with value will be returned in this module and facts module

**Comment by yuwzho** [comment]
ready_for_review

**Comment by Fred-sun** [comment]
@yungezz @zikalino Please help review this PR when you're available! Thanks!

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**